components/event-publisher-proxy/pkg/handler/nats_test.go

```diff
@@ -0,0 +1,205 @@
+	healthEndpoint := fmt.Sprintf("http://localhost:%d/healthz", port)
+	publishEndpoint := fmt.Sprintf("http://localhost:%d/publish", port)
+	publishLegacyEndpoint := fmt.Sprintf("http://localhost:%d/app/v1/events", port)
+	subscribedEndpointFormat := "http://localhost:%d/%s/v1/events/subscribed"
+	// wait that the embedded servers are started
+	testingutils.WaitForHandlerToStart(t, healthEndpoint)
+
+	// run the tests for publishing cloudevents
+	for _, testCase := range testCasesForCloudEvents {
+		t.Run(testCase.name, func(t *testing.T) {
+			body, headers := testCase.provideMessage()
+			resp, err := testingutils.SendEvent(publishEndpoint, body, headers)
+			if err != nil {
+				t.Errorf("Failed to send event with error: %v", err)
+			}
+			_ = resp.Body.Close()
+			if testCase.wantStatusCode != resp.StatusCode {
+				t.Errorf("Test failed, want status code:%d but got:%d", testCase.wantStatusCode, resp.StatusCode)
+			}
+		})
+	}
+	// run the tests for publishing legacy events
+	for _, testCase := range testCasesForLegacyEvents {
+		t.Run(testCase.name, func(t *testing.T) {
+			body, headers := testCase.provideMessage()
+
+			resp, err := testingutils.SendEvent(publishLegacyEndpoint, body, headers)
+			if err != nil {
+				t.Errorf("Failed to send event with error: %v", err)
+			}
+
+			if testCase.wantStatusCode != resp.StatusCode {
+				t.Errorf("Test failed, want status code:%d but got:%d", testCase.wantStatusCode, resp.StatusCode)
+			}
+
+			if testCase.wantStatusCode == http.StatusOK {
+				testingutils.ValidateOkResponse(t, *resp, &testCase.wantResponse)
+			} else {
+				testingutils.ValidateErrorResponse(t, *resp, &testCase.wantResponse)
+			}
+		})
+	}
+	// run the tests for subscribed endpoint
+	for _, testCase := range testCasesForSubscribedEndpoit {
+		t.Run(testCase.name, func(t *testing.T) {
+			subscribedURL := fmt.Sprintf(subscribedEndpointFormat, port, testCase.appName)
+			resp, err := testingutils.QuerySubscribedEndpoint(subscribedURL)
+			if err != nil {
+				t.Errorf("failed to send event with error: %v", err)
+			}
+
+			if testCase.wantStatusCode != resp.StatusCode {
+				t.Errorf("test failed, want status code:%d but got:%d", testCase.wantStatusCode, resp.StatusCode)
+			}
+			defer func() { _ = resp.Body.Close() }()
+			respBodyBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Errorf("failed to convert body to bytes: %v", err)
+			}
+			gotEventsResponse := subscribed.Events{}
+			err = json.Unmarshal(respBodyBytes, &gotEventsResponse)
+			if err != nil {
+				t.Errorf("failed to unmarshal body bytes to events response: %v", err)
+			}
+			if !reflect.DeepEqual(testCase.wantResponse, gotEventsResponse) {
+				t.Errorf("incorrect response, wanted: %v, got: %v", testCase.wantResponse, gotEventsResponse)
+			}
+		})
+	}
+}
+
+func newEnvConfig(port int) *env.NatsConfig {
+	return &env.NatsConfig{
+		Port:                  port,
+		NatsPublishURL:        fmt.Sprintf("http://localhost:%d", port+1),
```

Comment:
Why do we do `port+1`? This can get unmanageable in the future. Why `port` is not enough?

Reply:
The "Port" is the "ingress" port.
Nats server will run for these tests on the next available port: "port+1".
See also the "handler_test.go" for a similar solution.

Reply:
Changed